library/runtime_internal.h

```diff
@@ -99,9 +99,20 @@
 #undef MBEDTLS_AES_ACCELERATOR_NUM
 #endif /* MBEDTLS_AES_C */
 
-#if (defined(MBEDTLS_RUNTIME_C) || defined(MBEDTLS_CPU_HAS_FEATURES_ALT)) && \
+#if (defined(MBEDTLS_RUNTIME_C) || defined(MBEDTLS_CPUID_GET_ALT)) && \
     defined(MBEDTLS_AES_RUNTIME_HAVE_CODE)
 #define MBEDTLS_RUNTIME_HAVE_CODE
 #endif
 
+extern mbedtls_hwcap_mask_t mbedtls_cpu_hwcaps;
+static inline bool mbedtls_cpu_has_support(mbedtls_hwcap_mask_t mask)
+{
+    if (mbedtls_cpu_hwcaps == 0) {
+        mbedtls_cpu_hwcaps = mbedtls_cpuid_get();
+    }
+    return (mbedtls_cpu_hwcaps & mask) == mask;
+}
+#define MBEDTLS_RUNTIME_HAS_SUPPORT(mask)                   \
```

Comment:
If `mbedtls_cpu_has_support` is `static inline`, I don't think we need the macro.

I suggested this approach to follow how it's currently done in aesce (not static inline, use a macro to avoid function call overhead), but having it be `static inline` should do the same job.

Reply:
Just made a quick test - as it is now in e2e4aa3, this PR does not change code-size (as reported by `mtest -S`)

Reply:
That seems good - I am not sure I would expect a saving from this, but it shouldn't increase either.

Reply:
Will change it after I got CI result.